### DISCUSSION TITLE
i can't find data by it's slug by livewire

### DISCUSSION THREAD

**Discussion by muhammedn** [livewire/livewire#4949] [1. Help] [answered]
i create bookmark feature with livewire it was working good when using id ,but when i turned to slugs its stop working 

this my component controller

`public $recipe, $bookmarked;

    public function mount(Recipe $recipe){

        $this->recipe = $recipe;
        $this->bookmarked = $recipe->isBookmarked();

    }

    public function bookmark(Recipe $recipe){

        $message = '';

        if($recipe->isBookmarked()){

            $bookmark = Bookmark::where('recipe_id' , $recipe->id)->where('user_id', auth('web')->user()->id)->first();

            $bookmark->delete();

            $message = $recipe->title . ' removed from bookmark list successfully';
            $this->bookmarked = 0;
        }else{
            Bookmark::create([
                'user_id' => auth('web')->user()->id,
                'recipe_id' => $recipe->id,
            ]);

            $message = $recipe->title . ' bookmarked successfully';

            $this->bookmarked = 1;

        }
    }

    public function render()
    {
        return view('livewire.bookmarks-component');
    }` 
    
    and this my component 
    
    `<li>
    <a wire:click="bookmark({{$recipe->slug}})" title="@lang('content.add to favorite')" class="cursor-pointer flex flex-row items-center p-2 text-xs hover:bg-sky-50">
        <svg xmlns="http://www.w3.org/2000/svg" class="w-5 h-5 @if($bookmarked) fill-rose-500 stroke-rose-500 @endif" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
            <path stroke-linecap="round" stroke-linejoin="round" d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z" />
        </svg>

        @if(Request::is('*recipes'))
            @if($bookmarked )
                <span class="ms-2">@lang('content.remove from favorite')</span>
            @else
                <span class="ms-2">@lang('content.add to favorite')</span>
            @endif
        @endif
    </a>
</li>
`

it was working fine with id , when i used slugs stopped working

**Comment by bkilshaw** [comment, marked as answer]
I believe you'll need to wrap the slugs in quotes here:
```
wire:click="bookmark('{{$recipe->slug}}')"
```
Second, the default routeKey (the column that's used to find a record) is the `id` column. If you plan on using the slug instead, you'll need to set a custom key on your Recipe model:
```php
class Recipe {
  public function getRouteKeyName()
  {
      return 'slug';
  }
}
```

**Reply by muhammedn** [reply]
it fixed , thank you , I don't know how i missed to check this 😅
